App/View Controllers/Posts/PostsPageView.swift

```diff
@@ -327,15 +400,135 @@ final class PostsPageView: UIView {
         toolbar.isTranslucent = Theme.defaultTheme()[bool: "tabBarIsTranslucent"] ?? false
 
         topBar.themeDidChange(Theme.defaultTheme())
+        
+        // Update safe area gradient colors for new theme
+        configureSafeAreaGradient()
     }
-
+    
+    // MARK: - Safe Area Gradient Configuration
+    
+    private func configureSafeAreaGradient() {
+        // Remove existing gradient layer if any
+        safeAreaGradientView.layer.sublayers?.removeAll()
+        
+        let gradientLayer = CAGradientLayer()
+        let isDarkMode = Theme.defaultTheme()[string: "mode"] == "dark"
+        
+        if isDarkMode {
+            // Black to clear gradient for dark themes
+            gradientLayer.colors = [
+                UIColor.black.cgColor,
+                UIColor.black.withAlphaComponent(0.8).cgColor,
+                UIColor.black.withAlphaComponent(0.4).cgColor,
+                UIColor.clear.cgColor
+            ]
+            // Gradient locations - stronger at the top, fade to clear
+            gradientLayer.locations = [0.0, 0.3, 0.7, 1.0]
+        } else {
+            // For light mode, use a very subtle white gradient that blends seamlessly
+            gradientLayer.colors = [
+                UIColor.white.withAlphaComponent(0.8).cgColor,
+                UIColor.white.withAlphaComponent(0.6).cgColor,
+                UIColor.white.withAlphaComponent(0.2).cgColor,
+                UIColor.white.withAlphaComponent(0.02).cgColor,
+                UIColor.clear.cgColor
+            ]
+            // Fade more quickly to maintain subtlety
+            gradientLayer.locations = [0.0, 0.4, 0.7, 0.9, 1.0]
+        }
+        
+        gradientLayer.startPoint = CGPoint(x: 0.5, y: 0.0)
+        gradientLayer.endPoint = CGPoint(x: 0.5, y: 1.0)
+        
+        safeAreaGradientView.layer.addSublayer(gradientLayer)
+    }
```

Comment:
All of this could go into that UIView subclass, which could also implement themeDidChange.